packages/playwright-ct-react/registerSource.mjs

```diff
@@ -23,22 +23,48 @@ import { createRoot as __pwCreateRoot } from 'react-dom/client';
 /** @typedef {import('../playwright-test/types/experimentalComponent').Component} Component */
 /** @typedef {import('react').FunctionComponent} FrameworkComponent */
 
+/** @type {Map<string, () => Promise<FrameworkComponent>>} */
+const __pwLoaderRegistry = new Map();
+
 /** @type {Map<string, FrameworkComponent>} */
 const __pwRegistry = new Map();
+
 /** @type {Map<Element, import('react-dom/client').Root>} */
 const __pwRootRegistry = new Map();
 
 /**
- * @param {{[key: string]: FrameworkComponent}} components
+ * @param {{[key: string]: () => Promise<FrameworkComponent>}} components
  */
 export function pwRegister(components) {
   for (const [name, value] of Object.entries(components))
-    __pwRegistry.set(name, value);
+    __pwLoaderRegistry.set(name, value);
 }
 
 /**
  * @param {Component} component
  */
+async function __pwResolveComponent(component) {
+  if (typeof component !== 'object' || Array.isArray(component))
+    return
+
+  let componentFuncLoader = __pwLoaderRegistry.get(component.type);
+  if (!componentFuncLoader) {
+    // Lookup by shorthand.
```

Comment:
Carrying over my comment from the original PR: Now that you are resolving it here, do you want to remove it from the line 73 (new code)?

Reply:
Before I change everything, is this what you had in mind?: https://github.com/microsoft/playwright/pull/22494/commits/7755cc98043808917841e8834a4daa799001a20e

Reply:
Not exactly.

Reply:
Thank you for the feedback. Made another proposal: https://github.com/microsoft/playwright/pull/22494/commits/e8456360e8bd55e5601fd003cda4953078ebe01a what do you think?